.github/workflows/cpp_odbc.yml

```diff
@@ -0,0 +1,182 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: C++ ODBC
+
+on:
+  push:
+    branches:
+      - '**'
+      - '!dependabot/**'
+    tags:
+      - '**'
+    paths:
+      - '.github/workflows/cpp_odbc.yml'
+      - 'ci/scripts/cpp_*'
+      - 'cpp/src/arrow/flight/sql/odbc/*'
+  pull_request:
+    paths:
+      - '.github/workflows/cpp_odbc.yml'
+      - 'ci/scripts/cpp_*'
+      - 'cpp/src/arrow/flight/sql/odbc/*'
+  schedule:
+    - cron: '0 13 * * *'
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.head_ref || github.sha }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  windows:
+    runs-on: windows-2022
+    timeout-minutes: 240
+    env:
+      ARROW_BUILD_SHARED: ON
+      ARROW_BUILD_STATIC: ON
+      ARROW_BUILD_TESTS: ON
+      ARROW_BUILD_TYPE: release
+      ARROW_DEPENDENCY_SOURCE: VCPKG
+      ARROW_FLIGHT: ON
+      ARROW_FLIGHT_SQL: ON
+      ARROW_FLIGHT_SQL_ODBC: ON
+      # GH-47787 TODO Build ODBC installer
+      # ARROW_FLIGHT_SQL_ODBC_INSTALLER: ON
+      ARROW_SIMD_LEVEL: AVX2
+      CMAKE_CXX_STANDARD: "17"
```

Comment:
Should we use `20`?

Reply:
ODBC is using `CMAKE_CXX_STANDARD` 20, but I am not sure if Arrow Flight and Flight SQL support 20 yet, so I used 17 in the workflow to be safe.